spec/chingu/game_object_spec.rb

```diff
@@ -1,200 +1,216 @@
+# frozen_string_literal: true
+
 require 'spec_helper'
 
-module Chingu
-
-  describe GameObject do
-    before :each do
-      @game = Chingu::Window.new
-      
-      # Gosu uses the paths based on where rspec is, not where this file is, so we need to do it manually!
-      Gosu::Image::autoload_dirs.unshift File.join(File.dirname(File.expand_path(__FILE__)), 'images')
-    end
-
-    after :each do
-      @game.close
-    end
-
-    it { should respond_to(:x) }
-    it { should respond_to(:y) }
-    it { should respond_to(:angle) }
-    it { should respond_to(:center_x) }
-    it { should respond_to(:center_y) }
-    it { should respond_to(:factor_x) }
-    it { should respond_to(:factor_y) }
-    it { should respond_to(:zorder) }
-    it { should respond_to(:mode) }
-    it { should respond_to(:color) }
-    it { should respond_to(:attributes) }
-    it { should respond_to(:draw) }
-    it { should respond_to(:draw_at) }
-    it { should respond_to(:draw_relative) }
-
-    context "when created with defaults" do
-      it "should have default values" do
-        subject.angle.should == 0
-        subject.x.should == 0
-        subject.y.should == 0
-        subject.zorder.should == 100
-        subject.factor_x.should == 1
-        subject.factor_y.should == 1
-        subject.center_x.should == 0.5
-        subject.center_y.should == 0.5
-        subject.mode.should == :default
-        subject.image.should == nil        
-        subject.color.to_s.should == Gosu::Color::WHITE.to_s
-        subject.alpha.should == 255
-      end
-      
-      it "should wrap angle at 360" do
-        subject.angle.should == 0
-        subject.angle += 30
-        subject.angle.should == 30
-        subject.angle += 360
-        subject.angle.should == 30
-      end
+describe Chingu::GameObject do
+  before do
+    @game = Chingu::Window.new
 
-      it "shouldn't allow alpha below 0" do
-        subject.alpha = -10
-        subject.alpha.should == 0
-      end
+    # Add images/ to the resources load path
+    Gosu::Image.autoload_dirs
+               .unshift(File.expand_path('images', File.dirname(__FILE__)))
```

Comment:
What do you think about prepend vs. unshift? I think prepend is more readable for people who do not live in Ruby (or Perl?) every day. Just an idea though.

Reply:
Gladly! I actually had no idea of that syntax of the `prepend` syntax. I can't say much about readability because I don't have much experience with other languages as of yet, however I kinda can agree because unshift looks confusing to me every time I see it, prepend has a more direct ring to it.